src/manager/manager.c

```diff
@@ -83,6 +83,7 @@ void manager_unref(Manager *manager) {
         Node *node = NULL;
         Node *next_node = NULL;
         LIST_FOREACH_SAFE(nodes, node, next_node, manager->nodes) {
+                node_shutdown(node);
                 node_unref(node);
         }
         LIST_FOREACH_SAFE(nodes, node, next_node, manager->anonymous_nodes) {
```

Comment:
Don't you also need to shutdown anonymous nodes?

Reply:
I thought we can't send requests to anonymous nodes anyway, so there is no need to do that.
On the other hand maybe they'll need some other handling in the future, so it makes sense to call shutdown for them as well